MSCL/source/mscl/MicroStrain/MIP/MipNode_Impl.cpp

```diff
@@ -2191,6 +2191,29 @@ namespace mscl
         return r.parseResponse(doCommand(r, ActivationCode::buildCommand_get()));
     }
 
+    EventSupportInfo MipNode_Impl::getEventInfo(const EventSupportInfo::Query query) const
+    {
+        const MipFieldValues response = get(MipTypes::Command::CMD_EVENT_SUPPORT,
+                    { Value::UINT8(static_cast<uint8>(query)) }
+        );
+
+        EventSupportInfo info{};
+
+        info.query = static_cast<EventSupportInfo::Query>(response[0].as_int8());
+        info.maxInstances = response[1].as_int8();
+
+        const uint8 count = response[2].as_int8();
+        for (uint8 entryNumber = 0; entryNumber < count; entryNumber++)
+        {
+            const uint8 index = entryNumber * 2 + 3; // two values per element, first value count
```

Comment:
So I see how this line works, but it was confusing at first glance. To me, the following would be more clear:

```C
// two values per element, starting at the 4th byte
for (uint8 index = 3; index < count * 2 + 3; index += 2)
```

But actually now that I have written that out, it also seems not super clear. Keep whatever you think is more clear.

Reply:
I think your implementation is a bit cleaner. I also added a comment to try and clarify what's happening